smarts/core/remote_agent.py

```diff
@@ -80,24 +82,36 @@ def __init__(
 
     def act(self, obs):
         """Call the agent's act function asynchronously and return a Future."""
-        self._act_future = self._worker_stub.act.future(
+        self._grpc_future = self._worker_stub.act.future(
             worker_pb2.Observation(payload=cloudpickle.dumps(obs))
         )
 
-        return self._act_future
+        def get_action_future():
+            action = cloudpickle.loads(
+                        self._grpc_future.result().action
+                    )   
+            return action
+         
+        future = self._action_executor.submit(get_action_future)
```

Comment:
@Adaickalavan This was done to resolve the `action` so that it is the result given from `RemoteAgent.act(obs).result()`:

```python
# old
action = cloudpickle.loads(remote_agent.act(obs).result().action)
# new
action = remote_agent.act(obs).result()
```

I am not sure if you know of an easier way to do that which does not require wrapping the grpc `Future` in another `Future`.